From 27ed2b89b0d0887f37a5ecdc2b19525b8bba057e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Thu, 10 Nov 2022 10:37:00 +0000
Subject: [PATCH 10/11] [openmp] [test] Set the right calling convention for
 the Windows thread start function

This is required on i386 Windows; this fixes 99 testcases in that
build configuration.

Differential Revision: https://reviews.llvm.org/D137776
---
 openmp/runtime/test/omp_testsuite.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/openmp/runtime/test/omp_testsuite.h b/openmp/runtime/test/omp_testsuite.h
index d107c6159d6b..ef2ae20ec58c 100644
--- a/openmp/runtime/test/omp_testsuite.h
+++ b/openmp/runtime/test/omp_testsuite.h
@@ -86,7 +86,7 @@ struct thread_func_info_t {
 };
 
 // call the void* start_routine(void*);
-static DWORD __thread_func_wrapper(LPVOID lpParameter) {
+static DWORD WINAPI __thread_func_wrapper(LPVOID lpParameter) {
   struct thread_func_info_t* function_information;
   function_information = (struct thread_func_info_t*)lpParameter;
   function_information->start_routine(function_information->arg);
-- 
2.25.1

